lib/documents.js

```diff
@@ -2581,7 +2587,11 @@ function onQueryUrisRequest(output) {
   jobState.startTime = Date.now();
   jobState.endpoint = endpoint;
 
-  if(jobState.jobOptions.consistentSnapshot){
+  if(jobState.jobOptions.consistentSnapshot && (typeof jobState.jobOptions.consistentSnapshot === 'object')){
```

Comment:
A Boolean object would pass this test but not a test using instanceof mlutil.Timestamp

The same comment applies to the other initial validation.